src/evox/problems/classic/dtlz.py

```diff
@@ -217,75 +217,18 @@ def evaluate(self, state: chex.PyTreeDef, X: chex.Array):
         n, d = jnp.shape(X)
         m = self.m
         f = jnp.zeros((n, m))
-        g = 1 + jnp.mean(X[:, m-1:], axis=1, keepdims=True)
-        f = f.at[:, m-1].set((1 + g) * (m - jnp.sum(X[:, :m-1] / (1 + jnp.tile(g, (1, m-1)))
-                                                    * (1 + jnp.sin(3 * jnp.pi * X[:, :m-1])), axis=1, keepdims=True)))
-
-        return f, state
-
-    def pf(self, state: chex.PyTreeDef):
-        intervel = jnp.array([0, 0.251412, 0.631627, 0.859401])
-        median = (intervel[1] - intervel[0]) / (intervel[3] -
-                                                intervel[2] + intervel[1] - intervel[0])
-        # ReplicatePoint function
-        return super().pf(state)
-
-
-class DTLZ8(DTLZ):
-    def __init__(self, d=None, m=None, ref_num=1000):
-        if m is None:
-            self.m = 2
-        else:
-            self.m = m
-
-        if d is None:
-            self.d = self.m*10
-        else:
-            self.d = d
-        super().__init__(d, m, ref_num)
-
-    def evaluate(self, state: chex.PyTreeDef, X: chex.Array):
-        n, d = jnp.shape(X)
-        m = self.m
-        f = jnp.zeros((n, m))
-        for i in range(m):
-            f = f.at[:, i].set(
-                jnp.mean(X[:, int(i*d/m):int((i+1)*d/m)], axis=1, keepdims=True))
+        g = 1 + 9 * jnp.mean(X[:, m - 1:], axis=1, keepdims=True)
+        f = f.at[:, :m - 1].set(X[:, :m - 1])
+        f = f.at[:, m - 1:].set((1 + g) * (m - jnp.sum(f[:, :m - 1] / (1 + jnp.tile(g, (1, m - 1)))
+                                                       * (1 + jnp.sin(3 * jnp.pi * f[:, :m - 1])), axis=1,
+                                                       keepdims=True)))
 
         return f, state
 
     def pf(self, state: chex.PyTreeDef):
-        return super().pf(state)
-
-
-class DTLZ9(DTLZ):
-    def __init__(self, d=None, m=None, ref_num=1000):
-        if m is None:
-            self.m = 2
-        else:
-            self.m = m
-
-        if d is None:
-            self.d = self.m*10
-        else:
-            self.d = d
-        self.d = int(jnp.ceil(self.d/self.m)*self.m)
-        super().__init__(d, m, ref_num)
-
-    def evaluate(self, state: chex.PyTreeDef, X: chex.Array):
-        n, d = jnp.shape(X)
-        X = X**0.1
-        m = self.m
-        f = jnp.zeros([n, m])
-        for i in range(m):
-            f = f.at[:, i].set(jnp.sum(X[:, int(i*d/m):int((i+1)*d/m)],
-                                       axis=1, keepdims=True))
-        return f, state
-
-    def pf(self, state: chex.PyTreeDef):
-        n = self.ref_num * self.m
-        temp = jnp.vstack(
-            (jnp.hstack(((jnp.arange(0, 1, 1. / (n - 1))), 1.)))).T
-        f = jnp.c_[jnp.tile(jnp.cos(0.5*jnp.pi*temp),
-                            (1, self.m-1)), jnp.sin(0.5*jnp.pi*temp)]
-        return f, state
+        script_path = os.path.abspath(__file__)
```

Comment:
Use `[pkgutil](https://docs.python.org/3/library/pkgutil.html)` if you want to load static files. Manually loading static files has many downsides, for exmaple, it doesn't work if the code is published in a zip file.

Reply:
Done.